url882f91ec97_WenboCao_Microsoft_Drone_tgz-pJ8-org_ejml_alg_dense_decomposition_qr_StabilityQRDecompositionJ8/src/org/ejml/alg/dense/linsol/qr/LinearSolverQrpHouseCol.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2009-2014, Peter Abeles. All Rights Reserved.
+ *
+ * This file is part of Efficient Java Matrix Library (EJML).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ejml.alg.dense.linsol.qr;
+
+import org.ejml.alg.dense.decomposition.TriangularSolver;
+import org.ejml.alg.dense.decomposition.qr.QRColPivDecompositionHouseholderColumn_D64;
+import org.ejml.alg.dense.decomposition.qr.QrHelperFunctions;
+import org.ejml.data.DenseMatrix64F;
+
+/**
+ * <p>
+ * Performs a pseudo inverse solver using the {@link org.ejml.alg.dense.decomposition.qr.QRColPivDecompositionHouseholderColumn_D64} decomposition
+ * directly.  For details on how the pseudo inverse is computed see {@link BaseLinearSolverQrp}.
+ * </p>
+ * 
+ * @author Peter Abeles
+ */
+public class LinearSolverQrpHouseCol extends BaseLinearSolverQrp {
+
+    // Computes the QR decomposition
+    private QRColPivDecompositionHouseholderColumn_D64 decomposition;
+
+    // storage for basic solution
+    private DenseMatrix64F x_basic = new DenseMatrix64F(1,1);
+
+    public LinearSolverQrpHouseCol(QRColPivDecompositionHouseholderColumn_D64 decomposition,
+                                   boolean norm2Solution)
+    {
+        super(decomposition,norm2Solution);
+        this.decomposition = decomposition;
+    }
+
+    @Override
+    public void solve(DenseMatrix64F B, DenseMatrix64F X) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.